app/notifiers/course/video_notifier.rb

```diff
@@ -0,0 +1,24 @@
+class Course::VideoNotifier < Notifier::Base
+  def video_opening(user, video)
+    create_activity(actor: user, object: video, event: :opening).
+      notify(video.course, :email).
+      save!
+  end
+
+  def video_closing(user, video)
+    recipients = unattempted_users(video)
+
+    activity = create_activity(actor: user, object: video, event: :closing)
+    recipients.each { |r| activity.notify(r, :email) }
+    activity.save!
+  end
+
+  private
+
+  def unattempted_users(video)
+    course_users = video.course.course_users
+    students = course_users.student.without_phantom_users.includes(:user).map(&:user)
```

Comment:
Actually further thought about it, I think phantom users should also receive notification. The only different part of phantom users should be that they cannot seen by other users.